### ISSUE TITLE
mongodb - metrics exporter readiness and liveness probes fail with mongodb-exporter:0.33.x

### ISSUE THREAD

**Issue by rmuehlbauer** [bitnami/charts#11273]
### Name and Version

bitnami/mongodb:12.1.27

### What steps will reproduce the bug?

1. Install mongodb chart and enable metrics exporter - `metrics.enabled: true`
Problems seems to be bitnami/mongodb-exporter dependent - 0.33.x does not work whereas 0.32.x does work.
2. metrics container does not come up - Readiness probe failed: HTTP probe failed with statuscode: 404

### What is the expected behavior?

metrics container should startup. 
this problem seems to be somehow version dependent - bitnami/mongodb-exporter versions 0.32.x are working without problems. 
this issue only occours if you are using the latest 0.33.x version of the exporter

### What do you see instead?

```
  metrics:
    Container ID:  containerd://19de7ab1c1a7455e18397ab6427760804294032e6d70f1a5c343ef4298669356
    Image:         docker.io/bitnami/mongodb-exporter:0.33.0-debian-11-r3
    Image ID:      docker.io/bitnami/mongodb-exporter@sha256:c0351cdd43c4a4164662b542cc70f9d62a2732c3a1003602d3ceee2c75e59ef2
    Port:          9216/TCP
    Host Port:     0/TCP
    Command:
      /bin/bash
      -ec
    Args:
      /bin/mongodb_exporter --collect-all --compatible-mode --web.listen-address ":9216" --mongodb.uri "mongodb://$MONGODB_ROOT_USER:$(echo $MONGODB_ROOT_PASSWORD | sed -r "s/@/%40/g;s/:/%3A/g")@localhost:27017/admin?tls=true&tlsCertificateKeyFile=/certs/mongodb.pem&tlsCAFile=/certs/mongodb-ca-cert"

    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Wed, 20 Jul 2022 11:03:26 +0200
      Finished:     Wed, 20 Jul 2022 11:03:51 +0200
    Ready:          False
    Restart Count:  4
    Limits:
      memory:  256Mi
    Requests:
      cpu:      100m
      memory:   64Mi
    Liveness:   http-get http://:metrics/ delay=15s timeout=5s period=5s #success=1 #failure=3
    Readiness:  http-get http://:metrics/ delay=5s timeout=1s period=5s #success=1 #failure=3
    Environment:
      MONGODB_ROOT_USER:      root
      MONGODB_ROOT_PASSWORD:  <set to the key 'mongodb-root-password' in secret 'b2b-dev-mongodb-credentials'>  Optional: false
    Mounts:
      /certs from certs (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-b69lb (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  datadir:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  datadir-mongodb-b2b-dev-2
    ReadOnly:   false
  common-scripts:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      mongodb-b2b-dev-common-scripts
    Optional:  false
  config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      mongodb-b2b-dev
    Optional:  false
  scripts:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      mongodb-b2b-dev-scripts
    Optional:  false
  certs:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  certs-volume:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  mongodb-b2b-dev-ca
    Optional:    false
  kube-api-access-b69lb:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                   From                                   Message
  ----     ------     ----                  ----                                   -------
  Normal   Scheduled  2m27s                 gke.io/optimize-utilization-scheduler  Successfully assigned b2b/mongodb-b2b-dev-2 to gke-youplant-dev-dev-pool-1-70946474-wlwb
  Normal   Pulled     2m19s                 kubelet                                Container image "docker.io/bitnami/nginx:1.23.1-debian-11-r0" already present on machine
  Normal   Created    2m19s                 kubelet                                Created container generate-tls-certs
  Normal   Started    2m19s                 kubelet                                Started container generate-tls-certs
  Normal   Pulled     2m18s                 kubelet                                Container image "docker.io/bitnami/mongodb:5.0.9-debian-11-r7" already present on machine
  Normal   Created    2m18s                 kubelet                                Created container mongodb
  Normal   Started    2m18s                 kubelet                                Started container mongodb
  Normal   Pulled     112s (x2 over 2m18s)  kubelet                                Container image "docker.io/bitnami/mongodb-exporter:0.33.0-debian-11-r3" already present on machine
  Normal   Created    112s (x2 over 2m18s)  kubelet                                Created container metrics
  Normal   Started    112s (x2 over 2m18s)  kubelet                                Started container metrics
  Warning  Unhealthy  112s (x3 over 2m2s)   kubelet                                Liveness probe failed: HTTP probe failed with statuscode: 404
  Normal   Killing    112s                  kubelet                                Container metrics failed liveness probe, will be restarted
  Warning  Unhealthy  102s (x9 over 2m12s)  kubelet                                Readiness probe failed: HTTP probe failed with statuscode: 404
```

**Comment by fmulero** [comment]
Hi @rmuehlbauer 

This issue should be solved in #11398 Have you tried with that version?

**Comment by rmuehlbauer** [comment]
Hi @fmulero

yes its working again. the latest metric exporter version solved the issue - thank you very much!
